aii-ks/src/main/perl/ks.pm

```diff
@@ -427,12 +427,31 @@ timezone --utc $tree->{timezone}$ntp_servers
 rootpw --iscrypted $tree->{rootpw}
 EOF
 
+    my $repos = get_repos($config);
+    # error reported in get_repos
+    return if ! $repos;
+
     foreach my $url (@{$tree->{repo} || []}) {
         if ($url =~ m/^@(.+)$/) {
+            # find at least one repo with matching name
+            my $pattern = $1;
+            my @matches = grep {m/$pattern/} sort keys %$repos;
+            if (@matches) {
+                foreach my $reponame (@matches) {
+                    print "repo";
+                    foreach my $key (qw(name baseurl proxy includepkgs excludepkgs)) {
+                        my $val = $repos->{$reponame}->{$key};
+                        print " --$key=". (ref($val) eq 'ARRAY' ? join(',', @$val) : $val) if defined($val);
```

Comment:
If I'm not mistaken, this string should be filtered through `proxy_url()` to apply the global proxy settings, because we depend on that instead of having the proxy definitions repeated for every repository.

Reply:
no, the `get_repos` call does this already.

Reply:
but it does it badly. also using `proxy_url` there